controllers/openstackcluster_controller.go

```diff
@@ -709,8 +718,55 @@ func filterSubnets(networkingService *networking.Service, openStackCluster *infr
 				}
 				return nil, err
 			}
-			subnets = networkingService.ConvertOpenStackSubnetToCAPOSubnet(subnets, filteredSubnet)
+			filteredSubnets = append(filteredSubnets, *filteredSubnet)
 		}
 	}
-	return subnets, nil
+	return filteredSubnets, nil
+}
+
+// convertOpenStackNetworkToCAPONetwork converts an OpenStack network to a capo network.
```

Comment:
more of an code architecture question (ping @mdbooth / @lentzi90), I would have put these functions into network.go instead, as I've seen it elsewhere. However I'm not sure if we have a clear line on that.

Reply:
I'm still convinced that we want to move these new functions into `network.go`.